### ISSUE TITLE
Thiessen polygons connects over the east river

### ISSUE THREAD

**Issue by jeremysze** [jeremysze/LPIS#9]
I was just looking at my powerpoint slides at the shapefile of the thiessen polygons that I've created with GeoDa. I realize a problem that these polygons are linked across the East river. So intersections in Staten island share a border with another intersection in brooklyn. 
![image](https://user-images.githubusercontent.com/6887005/52442411-01f79b00-2af1-11e9-95eb-ce5aa906056b.png)

I am wondering if we can apply a mask so that some of these connections are stopped.

**Comment by jhconning** [comment]
Right... Could you just 'clip' the polygons by the boro shapefile to trim
polygons that extend over water to just retain the part that falls on
land.  This guys seems to do something like that
https://www.youtube.com/watch?v=eos1nR-5Tio

but I'm not sure if that leaves you with still workable polygons

**Comment by jeremysze** [comment]
I'm going to try the clipping with geopandas first. But in the mean time, I've just started running my spatial regressions in Stata. Fingers crossed they will all run!

**Comment by jeremysze** [comment]
https://github.com/jeremysze/LPIS/blob/master/Python_Clip%20thiessen%20polygons.ipynb

Done!

**Comment by jhconning** [comment]
Nice...

**Comment by jeremysze** [comment]
Actually, it is more complicated because of the shapefile of NYC. There are all these islands that will intersect with the theissen polygon. I'm trying to do a spatial join with my points file which has 12,987 unique intersections. After sjoin, I am somehow left with 12983 clipped theissen polygons.

**Comment by jhconning** [comment]
I can't see the notebook from phone but one possible solution (if problem
is that ny map used for clipping has lots of islands as polygons) would be
to remove small polygons from that shapefile with a mask to only keep
polygons of an area advice some very threshold.  Something like:

mask =nymap.area > 100

newnymap = nymap(mask)

(Should be square brackets)

And then use that for clipping ..

But if your nymap extends over water to include those small islands in one
large polygon then won't help

**Comment by jeremysze** [comment]
I got the sjoin to work. But I think some of the issue comes from multi-part polygons. I'm trying to explode those multipart polygons, and then do an sjoin again.

![image](https://user-images.githubusercontent.com/6887005/52506821-c8d92c80-2bbd-11e9-8555-9fd6528f8847.png)

**Comment by jeremysze** [comment]
I just fixed the issue. I exploded the borough map, which has 5 multi-polygons into 85. Then, I do the intersection command. Then, I do the sjoin to pick out only the polygons has an intersection point within the polygon. 
There were 4 points that were dropped in the process. In order to find them, I opened up QGIS to see the shapefiles. They fell outside the official shapefile of NYC (yellow dots)

![image](https://user-images.githubusercontent.com/6887005/52507830-b4e2fa00-2bc0-11e9-8e4d-017b4672ae43.png)
